pkg/cmd/pr/merge/merge.go

```diff
@@ -445,17 +474,18 @@ func NewMergeContext(opts *MergeOptions) (*mergeContext, error) {
 	}
 
 	return &mergeContext{
-		opts:              opts,
-		pr:                pr,
-		cs:                opts.IO.ColorScheme(),
-		baseRepo:          baseRepo,
-		isTerminal:        opts.IO.IsStdoutTTY(),
-		httpClient:        httpClient,
-		merged:            pr.State == MergeStateStatusMerged,
-		deleteBranch:      opts.DeleteBranch,
-		crossRepoPR:       pr.HeadRepositoryOwner.Login != baseRepo.RepoOwner(),
-		autoMerge:         opts.AutoMergeEnable && !isImmediatelyMergeable(pr.MergeStateStatus),
-		localBranchExists: opts.CanDeleteLocalBranch && git.HasLocalBranch(pr.HeadRefName),
+		opts:               opts,
+		pr:                 pr,
+		cs:                 opts.IO.ColorScheme(),
+		baseRepo:           baseRepo,
+		isTerminal:         opts.IO.IsStdoutTTY(),
+		httpClient:         httpClient,
+		merged:             pr.State == MergeStateStatusMerged,
+		deleteBranch:       opts.DeleteBranch,
+		crossRepoPR:        pr.HeadRepositoryOwner.Login != baseRepo.RepoOwner(),
+		autoMerge:          opts.AutoMergeEnable && !isImmediatelyMergeable(pr.MergeStateStatus),
+		localBranchExists:  opts.CanDeleteLocalBranch && git.HasLocalBranch(pr.HeadRefName),
+		mergeQueueRequired: pr.IsMergeQueueEnabled,
```

Comment:
not sure if this is better as a method on mergeContext or as a field.

I'm calling `m.pr.IsInMergeQueue` above, for consistency it's probably best to pick one or the other strategy

Reply:
If I understand this correctly, if the base branch of a PR has merge queue enabled, then that PR can only be added to merge queue, and not merged by traditional means?

Reply:
correct!

https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/configuring-pull-request-merges/managing-a-merge-queue

and 

https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/incorporating-changes-from-a-pull-request/merging-a-pull-request-with-a-merge-queue

for more context